sdk/runtime/runner.go

```diff
@@ -10,17 +10,25 @@ import (
 // Invoke runs the passed test-case and reports the result.
 func Invoke(tc func(*RunEnv) error) {
 	runenv := CurrentRunEnv()
+	defer runenv.MetricsPusher.Push()
 	defer runenv.Close()
 
+	startGauge := NewGauge(runenv, "start_time", "time of plan start")
+	startGauge.SetToCurrentTime()
+	runenv.RecordStart()
+
+	endGauge := NewGauge(runenv, "end_time", "time of plan end")
 	errfile, err := runenv.CreateRawAsset("run.err")
 	if err != nil {
 		runenv.RecordCrash(err)
+		endGauge.SetToCurrentTime()
```

Comment:
You can use `defer` to have a single call to `endGauge.SetToCurrentTime()` instead of 7 - otherwise it is very easy for someone to forget to add it if a new case appears, once this code gets refactored in the future.